pysmt/test/test_rewritings.py

```diff
@@ -83,6 +83,60 @@ def test_prenex_examples(self):
                         ok = not logic.quantifier_free
                     self.assertTrue(ok)
 
+    @skipIfNoSolverForLogic(QF_AUFLIA)
+    def test_ackermannization_unary(self):
+        a,b = (Symbol(x, INT) for x in "ab")
+        f,g,h = (Symbol(x, FunctionType(INT, [INT])) for x in "fgh")
+        #f(g(h(a))) = f(g(h(b)))
+        formula1 = Not(Equals(
+                    Function(f,
+                             [Function(g,
+                                       [Function(h, [a])])
+                             ]),
+                    Function(f,
+                             [Function(g,
+                                       [Function(h, [b])])
+                             ]),
+
+                ))
+        formula2 = Equals(a, b)
+        formula = And(formula1, formula2)
+        self._verify_ackermannization(formula)
+
+    @skipIfNoSolverForLogic(QF_AUFLIA)
+    def test_ackermannization_binary(self):
+        a,b = (Symbol(x, INT) for x in "ab")
+        f,g = (Symbol(x, FunctionType(INT, [INT, INT])) for x in "fg")
+        h = Symbol("h", FunctionType(INT, [INT]))
+        #f(g(h(a))) = f(g(h(b)))
+        formula1 = Not(Equals(
+                    Function(f,
+                             [a, Function(g,
+                                       [a, Function(h, [a])])
+                             ]),
+                    Function(f,
+                             [b, Function(g,
+                                       [b, Function(h, [b])])
+                             ]),
+
+                ))
+        formula2 = Equals(a, b)
+        formula = And(formula1, formula2)
+        self._verify_ackermannization(formula)
+
+    def _verify_ackermannization(self, formula):
```

Comment:
If you pass the logic to this method, you could then also iterate of the `example_formulae` to perform the equisat test on all the ones having UFs that are quantifier-free